lib/src/rules/public_member_api_docs.dart

```diff
@@ -101,9 +104,21 @@ class Visitor extends GeneralizingAstVisitor {
   bool isOverridingMember(Declaration node) =>
       getOverriddenMember(node.element) != null;
 
+  /// Return true if the given node is declared in a compilation unit that is in
+  /// a `lib/` folder.
+  bool isDefinedInLib(AstNode node) {
+    CompilationUnit compilationUnit = node is CompilationUnit
+        ? node
+        : node.getAncestor((node) => node is CompilationUnit);
+    Uri uri = compilationUnit?.element?.source?.uri;
+
+    // TODO(devoncarew): This will have false positives.
+    return uri != null && uri.pathSegments.contains('lib');
+  }
+
   @override
   visitClassDeclaration(ClassDeclaration node) {
-    if (isPrivate(node.name)) {
+    if (isPrivate(node.name) || !isDefinedInLib(node)) {
```

Comment:
It would be faster to do this once in `visitCompilationUnit` and store the result in a field.

Reply:
Makes sense, will do.